routes/api/v1/tasks.js

```diff
@@ -0,0 +1,61 @@
+const express = require("express");
+const router = express.Router();
+
+// CONTROLLERS
+
+const { catchErrors } = require("../../../config/errorHandler");
+const {
+	getTasks,
+	addTask,
+	updateTask,
+	getMyTasks,
+	deleteTask,
+	getTaskAssignees,
+	getUserTasks,
+	updateTaskAssignee
+} = require("../../../controllers/task_controller");
+const { allAuth, coreAuth } = require("../../../middlewares/auth");
+const {
+	addTaskValidation,
+	updateTaskValidation,
+	updateTaskAssigneeValidation
+} = require("../../../middlewares/validations");
+
+router.get(
+	"/assignees/:taid",
+	catchErrors(coreAuth),
+	catchErrors(getTaskAssignees)
+);
+
+router.put(
+	"/assignees/:taid",
+	catchErrors(allAuth),
+	catchErrors(updateTaskAssigneeValidation),
+	catchErrors(updateTaskAssignee)
+);
+
+// FETCH TASKS
+router.get("/:gid", catchErrors(allAuth), catchErrors(getTasks));
+router.get("/:gid/my", catchErrors(allAuth), catchErrors(getMyTasks));
+router.get("/:gid/:uid", catchErrors(coreAuth), catchErrors(getUserTasks));
```

Comment:
This won't allow a member who is head of a group to get the list of user tasks for their group.